src/pages/index.js

```diff
@@ -0,0 +1,318 @@
+const isLikedByUser = (likes = []) =>
+  Array.isArray(likes) &&
+  likes.some((user) => user._id === "LOGGED_IN_USER_ID");
+
+const openModal = (modal) => {
+  modal.classList.add("modal_opened");
+  document.addEventListener("keydown", handleEscapeKey);
+};
+
+const closeModal = (modal) => {
+  modal.classList.remove("modal_opened");
+  document.removeEventListener("keydown", handleEscapeKey);
+};
+
+const handleEscapeKey = (event) => {
+  if (event.key === "Escape") {
+    const modal = document.querySelector(".modal_opened");
+    if (modal) closeModal(modal);
+  }
+};
+
+// ------------------- SELECTORS -------------------
+
+const selectors = {
+  profileEditButton: document.querySelector(".profile__edit-btn"),
+  cardAddButton: document.querySelector(".profile__add-btn"),
+  profileName: document.querySelector(".profile__name"),
+  profileDescription: document.querySelector(".profile__description"),
+  profileForm: document.querySelector(".modal__form"),
+  nameInput: document.querySelector("#profile-name-input"),
+  jobInput: document.querySelector("#profile-description-input"),
+  editProfileModal: document.querySelector("#edit-profile-modal"),
+  cardModal: document.querySelector("#add-card-modal"),
+  cardForm: document.querySelector("#add-card-form"),
+  cardTemplate: document.querySelector("#card-template").content,
+  cardsContainer: document.querySelector(".cards"),
+  previewModal: document.querySelector("#preview-modal"),
+  previewImage: document.querySelector(".modal__image"),
+  previewCaption: document.querySelector(".modal__caption"),
+  avatarModal: document.querySelector("#avatar-modal"),
+
+  avatarPic: document.querySelector("#avatar-pic"),
+};
+
+// ------------------- API INSTANCE -------------------
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "d19feaee-a024-4f2d-b0a9-cad223e76830",
+    "Content-Type": "application/json",
+  },
+});
+
+// ------------------- FETCH & RENDER DATA -------------------
+api
+  .getAppInfo()
+  .then(([userInfo, cards]) => {
+    console.log("Cards fetched:", cards);
+    selectors.profileName.textContent = userInfo.name;
+    selectors.profileDescription.textContent = userInfo.about;
+    selectors.avatarPic.src = userInfo.avatar;
+    cards.forEach((card) => renderCard(card, "append"));
+  })
+  .catch(console.error);
+
+// ------------------- FORM HANDLERS -------------------
+const resetFormAndButton = (form) => {
+  form.reset();
+  const inputList = Array.from(form.querySelectorAll(settings.inputSelector));
+  const submitButton = form.querySelector(settings.submitButtonSelector);
+  toggleButtonState(inputList, submitButton, settings);
+  resetValidation(form, settings);
+};
+
+const handleAvatarFormSubmit = (evt) => {
+  evt.preventDefault();
+  const avatarInput = document.querySelector("#profile-avatar-input").value;
+  const saveButton = avatarForm.querySelector(".modal__submit-btn");
```

Comment:
It’s better to find **constant DOM elements** only 1 time at the top of the file so as not to waste resources on searching them again and again when you call a method (function) because searching is a very **hard** operation for the browser engine

Please, find each save button only **once**